src/groupByKey.ts

```diff
@@ -2,6 +2,43 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+interface Student {
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
+}
+
+interface Book {
+  author: string;
+  country: string;
+  imageLink: string;
+  language: string;
+  pages: number;
+  title: string;
+  year: number;
+}
+
+export function groupByKey(
+  items: [],
+  key: string,
+): GroupsMap<Book[] | Student[]> {
```

Comment:
The return type `GroupsMap<Book[] | Student[]>` is not accurate. It should be a map of string keys to arrays of the generic type used for `items`. As it stands, it implies that each group could be an array of either `Book` or `Student` objects, which is not the intended functionality.